### PR TITLE
Add format action

### LINKED ISSUE #1357: Run `format` and `fix` on CI and commit directly
### Description

Similar to https://github.com/getsentry/sentry-cocoa/blob/main/.github/workflows/format-code.yml

It should do for all the packages that run the analyzers.

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

- Add format actions which runs `dart format` & `dart fix` and commits the changes, if there are any.

## :bulb: Motivation and Context

Closes #1357

Ran both format and fix in package `flutter` & `dart` in this package. See commits.

Bildschirm­foto 2023-04-25 um 14 21 06

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        run: dart format .
+
+      - name: fix
+        run: dart fix --apply
+
+      # actions/checkout fetches only a single commit in a detached HEAD state. Therefore
+      # we need to pass the current branch, otherwise we can't commit the changes.
+      # GITHUB_HEAD_REF is the name of the head branch. GitHub Actions only sets this for PRs.
+      - name: Commit & Push
+        run: ../scripts/commit-formatted-code.sh $GITHUB_HEAD_REF
```
Comment (marandaneto) [inline .github/workflows/format.yml]: Can this actually be part of https://github.com/getsentry/sentry-dart/blob/main/.github/workflows/analyze.yml ?
Right now this is only executed by the dart and flutter packages, we have many more packages, so this approach would not work for all the packages.

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        package: [flutter, dart]
+    defaults:
+      run:
+        working-directory: ${{ matrix.package }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart@6a218f2413a3e78e9087f638a238f6b40893203d # pin@v1
+        if: ${{ matrix.package == 'dart' }}
+      - uses: subosito/flutter-action@48cafc24713cca54bbe03cdc3a423187d413aafa # pin@v2.10.0
+        if: ${{ matrix.package == 'flutter' }}
```
Comment (marandaneto) [inline .github/workflows/format.yml]: Do we actually need that? `format` and `fix` is part of the Dart SDK itself, maybe it does not depend on Flutter at all.

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: pub get
+        run: ${{ matrix.package }} pub get
+
+      - name: format
+        run: dart format .
```
Comment (marandaneto) [inline .github/workflows/format.yml]: Do we actually need to call `format`? doesn't `fix --apply` does it internally?

Comment (denrase) [reply .github/workflows/format.yml]: No, `fix --apply` handles issues uncovered by `dart analyze`, not the once by `dart format`

File: .github/workflows/format.yml (hunk 2 above)
Comment (denrase) [reply .github/workflows/format.yml]: Actually need it to run `flutter pub get`, but i guess this is not needed.

Comment (denrase) [reply .github/workflows/format.yml]: Correction, without running `dart/flutter pub get` the `fix` command removes needed imports.

Bildschirmfoto 2023-05-02 um 10 53 31

Bildschirmfoto 2023-05-02 um 10 53 35

**Comment by denrase** [comment]
@marandaneto Added the missing packages. We should probably still merge this with `analyze.yml`

File: .github/workflows/format.yml (hunk 1 above)
Comment (denrase) [reply .github/workflows/format.yml]: Added the other packages and tun `flutter pub get` where needed.

**Comment by marandaneto** [comment]
> @marandaneto Added the missing packages. We should probably still merge this with `analyze.yml`

Yes, asked about this [here](https://github.com/getsentry/sentry-dart/pull/1405#discussion_r1176562514).

**Comment by marandaneto** [comment]
@denrase After finishing this, we could copy-reuse the very same implementation for https://github.com/getsentry/sentry-dart-plugin

File: .github/workflows/analyze.yml [hunk 4]
```diff
@@ -36,8 +37,17 @@ jobs:
         if: ${{ inputs.sdk == 'dart' }}
       - uses: subosito/flutter-action@48cafc24713cca54bbe03cdc3a423187d413aafa # pin@v2.10.0
         if: ${{ inputs.sdk == 'flutter' }}
+
       - run: ${{ inputs.sdk }} pub get
-      - run: ${{ inputs.sdk }} format --set-exit-if-changed ./
+
+      - run: dart format .
+
+      - run: dart fix --apply
```
Comment (marandaneto) [inline .github/workflows/analyze.yml:45]: Should we still do `format --set-exit-if-changed ./` after this just for the sake of validity?

Comment (marandaneto) [reply .github/workflows/analyze.yml:45]: `dart fix` works but I am not sure if for 100% of the cases.

Comment (denrase) [reply .github/workflows/analyze.yml:45]: `dart fix` does not affect formatting AFAIK, it's just used for things linke unused imports and such, which gets detected by `dart analyze`. This i did leave in for exactly this reason, issues that `dart fix` cannot resolve. Formatting fixes will always work with just running `dart format .`

**Review by marandaneto — APPROVED** [review]
Thanks @denrase 
Lets do this after merging https://github.com/getsentry/sentry-dart/pull/1405#issuecomment-1534931856

**Pull request merged**